Update get-ovpn-files.sh

mbryde opened the pull request:
This will hopefully fix some of the issues with trying to redownloading the OVPN files over and over again. Now it should only do it when:
1. the file don't already exsist.
2. if the current file is older than two hours.
NOTE: This has not been tested directly yet. 

Should hopefully close these issues: 
Close #70
Close #49 
Close #42

Commit:
Update get-ovpn-files.sh

This will hopefully fix some of the issues with trying to redownloading the OVPN files over and over again. Now it should only do it when:
1. the file don't already exsist.
2. if the current file is older than two hours.

gerwim commented on app/ovpn/get-ovpn-files.sh:
```diff
@@ -2,26 +2,42 @@
-#Curl download ovpn files from NordVPN
-curl -s -o ${OVPN_CONFIG_DIR}/ovpn.zip ${OVPN_FILES}
+  #Create directory if no volume is done
+  mkdir -p /app/ovpn/config
```
The env var `OVPN_CONFIG_DIR` is `/app/ovpn/config` (which is mounted), so this should either be `mkdir -p ${OVPN_CONFIG_DIR}` or removed altogether because the path is mounted and therefore should exist. I know it's code which is present in the original source. :-)

Joentje replied:
To create the folder is valid when there is no mount. So when you run the image with no binding it will fail. I agree on your suggestion of using the env var 👍

gerwim commented:
>  NOTE: This has not been tested directly yet.

I have merged it into my fork, will test it the coming days ;-).

gerwim commented:
Three minor issues, see: https://github.com/gerwim/nordvpn-proxy/commit/5c83b41c8a01c32eec8f2f9d5e78a731865a6a23

Joentje commented:
@MBryde nice contribution 👍 did you have a reason for the 2 hours?

mbryde commented:
> @MBryde nice contribution 👍 did you have a reason for the 2 hours?

No not really. Could be less or more. Or even a variable you could set but default too something. It more or less up too you as the owner of the repo.

Commit:
Update get-ovpn-files.sh

Changes the mkdir too use the config dir variable. And some other small changes suggested by @gerwim

mbryde commented:
> Three minor issues, see: [gerwim@5c83b41](https://github.com/gerwim/nordvpn-proxy/commit/5c83b41c8a01c32eec8f2f9d5e78a731865a6a23)

I have changed the suggestions you made. Thank you for pointing it out. I'm kind of new to bash

Commit:
Refreshing files as a variable

Made it possible for people to change the amount of time inbetween the script will try and redownload the OVPN files.

Commit:
Update Dockerfile

mbryde commented:
> @MBryde nice contribution 👍 did you have a reason for the 2 hours?

I made it a variable you can change. But defaults to 120 minutes.

MBryde commented:
> @MBryde nice contribution 👍 did you have a reason for the 2 hours?

I made it a variable you can change. But defaults to 120 minutes.

MBryde commented:
> @MBryde nice contribution 👍 did you have a reason for the 2 hours?

I made it a variable you can change. But defaults to 120 minutes.

MBryde commented:
> @MBryde nice contribution 👍 did you have a reason for the 2 hours?

I made it a variable you can change. But defaults to 120 minutes.

mbryde commented:
@gerwim Did you have time to test it yet?

gerwim commented:
@MBryde I did. Just restarted the container, which comes back with:

```
/app/ovpn/get-ovpn-files.sh: line 9: download_files: command not found
```

I didn't debug the issue yet.

mbryde commented:
> @MBryde I did. Just restarted the container, which comes back with:
> 
> ```
> /app/ovpn/get-ovpn-files.sh: line 9: download_files: command not found
> ```
> 
> I didn't debug the issue yet.

Just read up on it a bit. And it's because the function is called before it's declared. Will move the function up before it's called. That should hopefully fix it.

Commit:
Update get-ovpn-files.sh

Moved function up before it's called.

gerwim commented on app/ovpn/get-ovpn-files.sh:
```diff
@@ -2,19 +2,6 @@
-  download_files
-fi
-
 function download_files() {
```
It should either be
```
download_files() {
```
or
```
function download_files {
```

See https://linuxize.com/post/bash-functions/

mbryde replied:
Should be fixed in the newest commit

Commit:
fixed download function

mbryde committed:
Added info to README file

Added the REFRESH_TIME env option to the readme file.

mbryde commented:
I have tested it myself. And it seems too work just fine. It won't re-download the files unless they either don't exists or they are older than the specified time. @Joentje if you could please test it now and close the RP that would be great.

Joentje commented:
Seems to be working fine! Great contribution! Thanks @gerwim @mbryde
